pilot/pkg/config/kube/gateway/deploymentcontroller.go

```diff
@@ -215,6 +232,20 @@ func (d *DeploymentController) Reconcile(req types.NamespacedName) error {
 		if !knownControllers.Contains(string(gc.Spec.ControllerName)) {
 			return nil
 		}
+		// find the tag or revision indicated by the object
```

Comment:
this should be outside the L230 `if`; there are other paths that allow getting to configureIstioGateway

Reply:
I'm not sure I follow.  If the gateway is of the correct revision, but not the correct class, we shouldn't reconcile it, right?

Reply:
L253 allows for it to be the correct class as well